c++/src/kj/compat/http.h

```diff
@@ -718,13 +718,24 @@ class HttpClient {
   // stack-allocated).
 
   struct ConnectRequest {
-    struct Status {
+    struct Status : public kj::Refcounted {
```

Comment:
Converting this struct into a heap-allocated refcounted type is kind of sad. I guess you did this to allow for `fork()`ing the promise, but as I noticed above, it doesn't look like the `fork()` is actually needed. Can we remove it and then turn this back into a value type?

Reply:
Yeah I'm not happy with this either. Once I finish working through adding errorBody back in I'm going to change this up. I'm pretty sure the forks aren't needed now.